pkg/services/object/get/util.go

```diff
@@ -222,72 +219,50 @@ func (c *clientWrapper) get(exec *execCtx, key *ecdsa.PrivateKey) (*object.Objec
 
 	hdr, rdr, err := c.client.ObjectGetInit(exec.context(), addr.Container(), id, user.NewAutoIDSigner(*key), opts)
 	if err != nil {
-		return nil, fmt.Errorf("init object reading:: %w", err)
+		return nil, nil, fmt.Errorf("init object reading:: %w", err)
 	}
-
-	buf := make([]byte, hdr.PayloadSize())
-
-	_, err = rdr.Read(buf)
-	if err != nil && !errors.Is(err, io.EOF) {
-		return nil, fmt.Errorf("read payload: %w", err)
-	}
-
-	hdr.SetPayload(buf)
-
-	return &hdr, nil
+	return &hdr, rdr, nil
 }
 
-func (e *storageEngineWrapper) get(exec *execCtx) (*object.Object, error) {
+func (e *storageEngineWrapper) get(exec *execCtx) (*object.Object, io.ReadCloser, error) {
 	if exec.headOnly() {
 		r, err := e.engine.Head(exec.address(), exec.isRaw())
 		if err != nil {
-			return nil, err
+			return nil, nil, err
 		}
 
-		return r, nil
+		return r, nil, nil
 	}
 
 	if rng := exec.ctxRange(); rng != nil {
 		r, err := e.engine.GetRange(exec.address(), rng.GetOffset(), rng.GetLength())
 		if err != nil {
-			return nil, err
+			return nil, nil, err
 		}
 
 		o := object.New()
 		o.SetPayload(r)
 
-		return o, nil
+		return o, nil, nil
 	}
 
-	header, reader, err := e.engine.GetStream(exec.address())
-	if err != nil {
-		return nil, err
-	}
-	defer func() { _ = reader.Close() }()
-
-	payload, err := io.ReadAll(reader)
-	if err != nil {
-		return nil, fmt.Errorf("can't read object payload: %w", err)
-	}
-	header.SetPayload(payload)
-	return header, nil
+	return e.engine.GetStream(exec.address())
 }
 
 func (w *partWriter) WriteChunk(p []byte) error {
 	return w.chunkWriter.WriteChunk(p)
 }
 
 func (w *partWriter) WriteHeader(o *object.Object) error {
+	// Range responses use only ChunkWriter and didn't emit headers.
+	// Treat header writes as a no-op when no header writer is set so callers can
+	// still "trigger" header-dependent flows without affecting the client stream.
+	if w.headWriter == nil {
```

Comment:
seems like previously this method was not called for RANGE, otherwise NPE would occur. May u pls explain why we need this now?